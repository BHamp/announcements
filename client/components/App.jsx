/* eslint-disable import/extensions */
import React, { useState, useEffect } from 'react';
import Announcements from './Announcements.jsx';
import Articles from './Articles.jsx';

const App = () => {
  const [showArticles, setShowArticles] = useState(false);

  const toggleArticles = () => {
    setShowArticles(!showArticles);
  };

  return (
    <div>
      <div id="Announcements">
        <Announcements toggleArticles={toggleArticles} />
      </div>
      <div id="article-modal">
        {(showArticles) ? <Articles toggleArticles={toggleArticles} /> : null}
      </div>
    </div>
  );
};

export default App;
