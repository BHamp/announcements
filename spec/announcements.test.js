/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable no-undef */
/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { mount, shallow, render } from 'enzyme';
import App from '../client/components/App';

// require('regenerator-runtime/runtime');

// const mongoose = require('mongoose');
// const supertest = require('supertest');

// const database = 'fec';
// const app = require('../index.js');

// const request = supertest(app); // this is now essentially the server for testing

// beforeAll(async () => {
//   const url = `mongodb://127.0.0.1/${database}`;
//   await mongoose.connect(url, { useNewUrlParser: true, useUnifiedTopology: true });
// });

describe('App Component', () => {
  it('should render the App component correctly', () => {
    const component = shallow(<App />);
    expect(component).toMatchSnapshot();
  });

  // it('should render the App component correctly with a given game', async (done) => {
  //   const res = await request.get('/randomGame');
  //   const randomGame = res.body;
  //   const component = shallow(<App game={randomGame} />);
  //   expect(component).toMatchSnapshot();
  //   done();
  // });
});
