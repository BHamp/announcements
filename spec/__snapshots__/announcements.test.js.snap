// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should render the App component correctly 1`] = `
<div>
  <div
    className="events"
    id="Announcements"
  >
    <Announcements
      game={Object {}}
      kind="event"
      toggleArticles={[Function]}
    />
  </div>
  <div
    className="announcements"
    id="Announcements"
  >
    <Announcements
      game={Object {}}
      kind="announcement"
      toggleArticles={[Function]}
    />
  </div>
  <div
    id="article-modal"
  />
</div>
`;

exports[`App Component should render the App component correctly with a given game 1`] = `
<div>
  <div
    className="events"
    id="Announcements"
  >
    <Announcements
      game={Object {}}
      kind="event"
      toggleArticles={[Function]}
    />
  </div>
  <div
    className="announcements"
    id="Announcements"
  >
    <Announcements
      game={Object {}}
      kind="announcement"
      toggleArticles={[Function]}
    />
  </div>
  <div
    id="article-modal"
  />
</div>
`;
